### PR TITLE
RSDK-2131 - Refactor generic component, add generic wrappers

### PR DESCRIPTION
**Pull request by stuqdog**

### CODE REVIEW

File: src/module/service.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 #include <boost/none.hpp>
 #include <common/utils.hpp>
 #include <components/component_base.hpp>
-#include <components/generic.hpp>
+#include <components/generic/generic.hpp>
```
Comment (cheukt) [inline src/module/service.cpp:27]: do we still need the generic import here?

Comment (stuqdog) [reply src/module/service.cpp:27]: nope! thanks for spotting

**Review by cheukt — APPROVED** [review]
lgtm, tho it does seem like some of the includes can be cleaned up

**Comment by stuqdog** [comment]
> lgtm, tho it does seem like some of the includes can be cleaned up

yeah, for sure that's the case :) I'm kinda lumping that in with the general refactoring work that needs to come, but will be cleaning it all up soon!

**Pull request merged**